File: Ozone_Route256_2023/Task_B/solution.go
// (B) Сумма к оплате (10 баллов)
// Полное решение: 10 баллов

package main
 
import (
	"bufio"
	"fmt"
	"os"
)
 
func main() {
	var a, b, t, sum int
	input := bufio.NewReader(os.Stdin)
	output := bufio.NewWriter(os.Stdout)
	defer output.Flush()
 
	fmt.Fscan(input, &a)
 
	for ; a > 0; a-- {
		basket := make(map[int]int)
		fmt.Fscan(input, &b)
		sum = 0
		// Read data from StdIn
		for ; b > 0; b-- {
			fmt.Fscan(input, &t)
			basket[t]++
		}
		// Calculate
		for price, amount := range basket {
			sum += (amount - amount/3) * price
		}
 
		fmt.Fprint(output, sum, "\n")
	

File: Ozone_Route256_2023/Task_B/solution.py
# (B) Сумма к оплате (10 баллов), Полное решение: 10 баллов.

def solution() -> int:
    basket = {}
    for product in map(int, input().split()):
        if product in basket:
            basket[product] += 1
            continue
        basket[product] = 1
    return sum(
        (amount - amount // 3) * prod for prod, amount in basket.items()
    )


def main():
    for _ in range(int(input())):
        input()
        print(solution())


if __name__ == '__main__':
    main()
